### ISSUE TITLE
typescript language server can't figure out angular import structure

### ISSUE THREAD

**Issue by mhamri** [angular/angular#52008]
### Which @angular/* package(s) are the source of the bug?

common

### Is this a regression?

No

### Description

opening #40407 again

Since it was mentioned that the issue was fixed after the recent changes, I'd like to point out that the problem still exists. I have attached a screen recording to demonstrate that the issue is not resolved. It is quite inconvenient to have to refer to the documentation for every required import. As previously mentioned, this issue has already been reported to the TypeScript team, and it appears to be related to the Angular export structure.

## vscode - standalone HttpClient
can't get any suggestion on any `provide*` 

https://github.com/angular/angular/assets/8991783/bb4eba31-ee60-4e54-ae89-44701a778251

## vscode - standalone - anything inside Material library

https://github.com/angular/angular/assets/8991783/99bc2586-9f7f-4d7b-be5e-5d46db16485a

## vscode - Module - HttpClient

https://github.com/angular/angular/assets/8991783/7c782da4-b54b-4963-ae22-7724cce69477

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Global setting: enabled
Local setting: disabled
Effective status: disabled

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 16.2.4
Node: 16.20.0
Package Manager: npm 8.19.4
OS: win32 x64

Angular: 16.2.7
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1602.4
@angular-devkit/build-angular   16.2.4
@angular-devkit/core            16.2.4
@angular-devkit/schematics      16.2.4
@angular/cli                    16.2.4
@schematics/angular             16.2.4
rxjs                            7.8.1
typescript                      5.1.6
zone.js                         0.13.3
```

**Comment by JoostK** [comment]
Angular packages have proper `exports` maps in the root `package.json` file, but these are only used by TypeScript for [modern resolution modules](https://www.typescriptlang.org/tsconfig#moduleResolution). In particular, `node16` and `bundler` should allow these to resolve. I believe there are currently reasons why `node16` cannot be adopted without major changes , but `bundler` should impose less restrictions and may allow this to work.

**Comment by JeanMeche** [comment]
Yeah, `moduleResolution: bundler` does it for you. Also mentionning that the CLI included a migration for it in v20: angular/angular-cli/pull/30000
